testing/src/uploadResult.ts

```diff
@@ -0,0 +1,59 @@
+import axios, { AxiosResponse } from "axios";
+import FormData from "form-data";
+import fs from "fs";
+
+const TQ_USERNAME = process.env.TQ_USER_NAME;
+const TQ_PASSWORD = process.env.TQ_PASSWORD;
+const TQ_SECRET = process.env.TQ_SECRET;
+const TQ_RUN_ID = process.env.TQ_RUN_ID; //( default id is 26606. Will need to change once we have integrated TQ account for the team)
+
+const getAuthToken = async (): Promise<AxiosResponse<any>> => {
+  const url = "https://api.testquality.com/api/oauth/access_token";
+  const body = JSON.stringify({
+    grant_type: "password",
+    client_id: "2",
+    client_secret: TQ_SECRET,
+    username: TQ_USERNAME,
+    password: TQ_PASSWORD,
+  });
+  const settings = {
+    headers: {
+      "Content-Type": "application/json",
+    },
+  };
+  try {
+    console.log("Fetching the token from TestQuality");
+    const response = await axios.post(url, body, settings);
+    console.log("Token Fetched");
+    return response.data["access_token"];
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+};
+
+async function uploadResultFile() {
+  const auth = await getAuthToken();
+  const url = `https://api.testquality.com/api/plan/${TQ_RUN_ID}/junit_xml`;
+  var formData = new FormData();
+  var resultFile = fs.createReadStream("test-results.xml");
+  formData.append("file", resultFile, "test-results.xml");
+  try {
+    if (String(auth).length > 1) {
```

Comment:
No need to do a check here, IMO. If the API call to get the access token is 200, the token will be there, is it fail any different HTTP code Axios will throw an exception.